pymongo/mongo_client.py

```diff
@@ -1408,12 +1408,17 @@ def is_retrying():
                 if retryable_error:
                     session._unpin()
                 if not retryable_error or (is_retrying() and not multiple_retries):
-                    raise
+                    if exc.has_error_label("NoWritesPerformed"):
+                        assert last_error is not None
+                        raise last_error from exc
+                    else:
+                        raise
```

Comment:
There's a missing condition here. If the current error is NoWritesPerformed and the last error is None, then we must raise the current error even though it's labelled NoWritesPerformed. This is True for all the cases above with `assert last_error is not None` too.